lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -1478,19 +1481,13 @@ namespace Js
             extractor->MarkVisitVar(this->result);
         }
 
-        if(this->resolveReactions != nullptr)
-        {
-            for(int32 i = 0; i < this->resolveReactions->Count(); ++i)
-            {
-                this->resolveReactions->Item(i)->MarkVisitPtrs(extractor);
-            }
-        }
-
-        if(this->rejectReactions != nullptr)
+        if(this->reactions != nullptr)
         {
-            for(int32 i = 0; i < this->rejectReactions->Count(); ++i)
+            JavascriptPromiseReactionList::Iterator it = this->reactions->GetIterator();
```

Comment:
nit: you could use Map function on SList, which is easier to understand.

Reply:
Updated in a couple of places, and left as-is in a couple others since the closures were more complex.